Anurse/eerhardt update dependencies

anurse opened the pull request:
Relocated from https://github.com/dotnet/cli/pull/1990 so I could add commits ;)

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/dotnet/cli/2040)
<!-- Reviewable:end -->

weshaggard commented:
The compile errors look correct. Currently Microsoft.Extensions.DependencyModel is targeting netstandard1.3 and in that generation System.Reflection.Assembly doesn't contain the GetEntryAssembly. That was added in netstandard1.5 version of System.Reflection. So that dependency looks like it will need to be tweaked.

anurse commented:
Yeah, I had changed it to `netstandard1.5` and then realized there was a merge issue and then when I fixed it I blasted away the change to 1.5 :). Fixed

anurse commented:
Windows 2016 was all that was left, so merged.
